src/macos.rs

```diff
@@ -258,6 +278,31 @@ fn dns_servers_from_etc_resolv_conf_file() -> std::io::Result<Vec<IpAddr>> {
     Ok(dns_servers)
 }
 
+fn get_dns_servers_if_exist_from_interface(original_gw_scope: &str) -> std::io::Result<()> {
+    let script = format!(
+        r#"
+    original_gw_scope={}
```

Comment:
Even if `original_gw_scope` is supposed to be an interface name, we should deal with that as untrusted input. What characters can such interface name contain? It would be better to implement that logic in Rust directly for security reasons.